opal/util/os_dirpath.c

```diff
@@ -112,14 +112,12 @@ int opal_os_dirpath_create(const char *path, const mode_t mode)
             strcat(tmp, parts[i]);
         }
 
-        /* Now that we finally have the name to check, check it.
-           Create it if it doesn't exist. */
-        ret = mkdir(tmp, mode);
-        if ((0 > ret && EEXIST != errno) || 0 != stat(tmp, &buf)) {
-            opal_output(0,
-                        "opal_os_dirpath_create: "
-                        "Error: Unable to create the sub-directory (%s) of (%s), mkdir failed [%d]\n",
-                        tmp, path, ret);
+        /* Now that we have the name, try to create it */
+        mkdir(tmp, mode);
+        ret = errno;  // save the errno for an error msg, if needed
+        if (0 != stat(tmp, &buf)) {
```

Comment:
the pointed commit was meant to solve a race condition (e.g. ```mkdir()``` after ```stat()```)
 
there still is a tricky race condition here ... someone else might have ```mkdir()``` but with the wrong permissions, so it will likely crash later.
so i guess we have to check the permissions at the end.

this is for master
```patch
diff --git a/opal/util/os_dirpath.c b/opal/util/os_dirpath.c
index 9ef9ebb..2328179 100644
--- a/opal/util/os_dirpath.c
+++ b/opal/util/os_dirpath.c
@@ -9,7 +9,7 @@
  *                         University of Stuttgart.  All rights reserved.
  * Copyright (c) 2004-2005 The Regents of the University of California.
  *                         All rights reserved.
- * Copyright (c) 2015-2016 Research Organization for Information Science
+ * Copyright (c) 2015-2017 Research Organization for Information Science
  *                         and Technology (RIST). All rights reserved.
  * Copyright (c) 2016-2017 Intel, Inc. All rights reserved.
  * $COPYRIGHT$
@@ -121,6 +121,12 @@ int opal_os_dirpath_create(const char *path, const mode_t mode)
             opal_argv_free(parts);
             free(tmp);
             return OPAL_ERROR;
+        } else if (i == (len-1) && (mode != (mode & buf.st_mode)) && (0 > chmod(tmp, (buf.st_mode | mode)))) {
+            opal_show_help("help-opal-util.txt", "dir-mode", true,
+                           tmp, mode, strerror(errno));
+            opal_argv_free(parts);
+            free(tmp);
+            return(OPAL_ERR_PERM); /* can't set correct mode */
         }
     }

```